pkg/daemon/update.go

```diff
@@ -255,6 +207,76 @@ func (dn *Daemon) compareMachineConfig(oldConfig, newConfig *mcfgv1.MachineConfi
 	return true, nil
 }
 
+// addExtensionsRepo adds a repo into /etc/yum.repos.d/ which we use later to
+// install extensions and rt-kernel
+func addExtensionsRepo() error {
+	repoContent := "[coreos-extensions]\nenabled=1\nmetadata_expire=1m\nbaseurl=" + osImageContentDir + "/extensions/\ngpgcheck=0\nskip_if_unavailable=False\n"
+	if err := writeFileAtomicallyWithDefaults(extensionsRepo, []byte(repoContent)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func extractOSImage(imgURL string) error {
+	var registryConfig []string
+	if _, err := os.Stat(kubeletAuthFile); err == nil {
+		registryConfig = append(registryConfig, "--registry-config", kubeletAuthFile)
+	}
+	if err := os.MkdirAll(osImageContentDir, 0755); err != nil {
+		return fmt.Errorf("error creating directory %s: %v", osImageContentDir, err)
+	}
+	// Extract the image
+	args := []string{"image", "extract", "--path", "/:" + osImageContentDir}
```

Comment:
In the future with https://github.com/openshift/machine-config-operator/pull/1947 I think we're going to want to wrap this in a `ionice -c 3` or so.  No need to do so in this patch, just noting.  This also argues for keeping the extraction in a separate process going forward (which could just be us re-exec'ing our own binary of course).

Reply:
will keep this in mind to fix in a follow-up PR